src/Elasticsearch.Net/Serialization/DiagnosticsSerializerProxy.cs

```diff
@@ -46,11 +47,23 @@ public DiagnosticsSerializerProxy(IElasticsearchSerializer serializer, string pu
 		{
 			_serializer = serializer;
 			_state = new SerializerRegistrationInformation(serializer.GetType(), purpose);
-			if (serializer is IInternalSerializerWithFormatter withFormatter)
-				_formatterResolver = withFormatter.FormatterResolver;
+			if (serializer is IInternalSerializer s && s.TryGetJsonFormatter(out var formatterResolver))
+			{
+				_formatterResolver = formatterResolver;
+				_wrapsUtf8JsonSerializer = true;
+			}
+			else
+			{
+				_wrapsUtf8JsonSerializer = false;
```

Comment:
minor nitpick: can the assignments be ordered the same in the `else` branch as per the `if` branch?

Reply:
++